File: Matlab/SCRoF_Fevrier/asPairesIndicatrices.m
function AS=asPairesIndicatrices(AS)
% AS=asPairesIndicatrices(AS);
% Identifier les variables indicatrices, s'il y en a pour la branche AS.branche
AS.Cpaires=nchoosek(AS.pertinent,2); % tous les sous-ensembes de 2 parmi les v
nc=size(AS.Cpaires,1);
AS.Crit=zeros(nc,1);
AS.Corr=zeros(AS.nv,nc);
AS.Ppaires=zeros(nc,1);
AS.doublet=[];
for k=1:nc
    AS=asAnnulePaire(AS,k);  % optimiser les variables de la rangée k de AS.Cpaires
end
AS.z2=(AS.Corr).^2*(AS.N-1);

File: R/asPairesIndicatrices.R
asPairesIndicatrices <- function(AS) {
  # Identifier les variables indicatrices, en vérifiant l'annulation de toutes les paires de variables non-orphelines
  # POC : length de AS$pertinent et non pas le contenu de AS$pertinent
  AS$Cpaires <- combn(length(AS$pertinent), 2) # tous les sous-ensembles de 2 parmi les variables pertinentes
  nc <- ncol(AS$Cpaires)
  AS$Crit <- rep(0, nc)      # max(abs(Corr))
  AS$Prob <- rep(0, nc)      # corrigé pour le nombre de corrélations
  AS$Ppaires <- rep(0, nc)   # les poids d'annulation
  AS$satPaires <- matrix(NA,nrow=2,ncol=nc)
  AS$Corr <- matrix(0, AS$nv-2-length(AS$orphelines), nc)
  AS$doublet <- NULL
  for (k in 1:nc) {
    AS <- asAnnulePaire(AS, k)  # optimiser les variables de la rangée k de AS$Cpaires
  }
  return(AS)
}
